<launch>
    <env name="GAZEBO_RESOURCE_PATH" value="$(find turtlebot3_gazebo)/models/autorace/ground_picture"/>

    <arg name="x_pos" default="0.8"/>
    <arg name="y_pos" default="-1.747"/>
    <arg name="z_pos" default="0"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find autorace_host)/worlds/turtlebot3_autorace_2020.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <param name="robot_description"
           command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger_for_autorace_2020.urdf.xacro"/>
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"
          args="-urdf -model autorace -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description"/>


    <!--  Intrinsic camera calibration-->
    <group>
        <!-- image transport from compressed to raw -->
        <node pkg="image_transport" type="republish" name="republish" args="compressed">
            <remap from="in" to="camera/rgb/image_raw"/>
            <remap from="out" to="camera/image"/>
        </node>
        <node pkg="topic_tools" type="relay" name="relay_camera_info"
              args="/camera/rgb/camera_info /camera/camera_info"/>
    </group>

    <!-- image rectification -->
    <node ns="camera" pkg="image_proc" type="image_proc" name="image_proc" args="_approximate_sync:=true"
          output="screen">
        <remap from="image_raw" to="image"/>
        <param name="queue_size" value="20"/>
    </node>

    <!--  Start mission-->
    <node pkg="turtlebot3_autorace_core" type="core_node_mission" name="core_node_mission" output="screen"/>
</launch>